test/hotspot/gtest/runtime/test_cgroupSubsystem_linux.cpp

```diff
@@ -34,6 +34,10 @@
 
 #include <stdio.h>
 
+#ifdef MUSL_LIBC
+#include <libgen.h> // for basename
+#endif
```

Comment:
Does glibc provide its own definition of `basename`? From what I can see on Linux `man -s 3 basename` shows:
```
SYNOPSIS
       #include <libgen.h>
..
       char *basename(char *path);
```
which seems to eb same as what you now have for MUSL-C. ???

Reply:
Good point. The man page also has:

```
VERSIONS
       There are two different versions of basename() - the POSIX version  de‐
       scribed above, and the GNU version, which one gets after

               #define _GNU_SOURCE         /* See feature_test_macros(7) */
               #include <string.h>

       The  GNU  version  never  modifies  its argument, and returns the empty
       string when path has a trailing slash, and in particular also  when  it
       is "/".  There is no GNU version of dirname().

       With glibc, one gets the POSIX version of basename() when <libgen.h> is
       included, and the GNU version otherwise.
```

So it sounds like we use the GNU version when `glibc` is being used.

Reply:
Yeah. Since we want to be libc agnostic, we should use the POSIX variant, though..

Reply:
Thanks @jerboaa I obviously completely missed that. :(

But as per Thomas's comment we should be using the Posix function not glibc version. Then we would not need to specialize for Musl-C.

Reply:
So should I remove the ifdef MUSL_LIBC  ?
I considered this but was not sure because  it might change more than just fixing the Alpine build.

Reply:
Yes, you can remove it. Both libc's actually require libgen.h for basename.